js/index8.js

```diff
@@ -548,15 +548,647 @@ console.log(uniqueSortedCourses); // ['biologie', 'informatică', 'limba englez
+}
+const add = (value) => value + 2;
+
+console.log(radu);
+
+console.log(getMap(radu, add));
+
+console.log(`Implement the following methods using a for loop:
+ - filter`);
+
+function getFilter(array, callback) {
+  let newArray = [];
+  for (let arr of array) {
+    if (callback(arr) === true) {
+      newArray.push(arr);
+    } else if (callback(arr) === false) {
+      continue;
+    }
+  }
+  return newArray;
+}
+const add0 = (value) => {
+  if (value > 100) {
+    return true;
+  }
+  return false;
+};
+
+console.log(add0(7));
+console.log(add0(71));
+console.log(add0(71) === true);
+
+console.log(radu);
+
+console.log(getFilter(radu, add0));
+
+console.log(`Implement the following methods using a for loop:
+ - find`);
+
+console.log(`Implement the following methods using a for loop:
+ - every`);
+
+console.log(`Implement the following methods using a for loop:
+ - some`);
+
+// - exercitiul-1 -
+console.log(`- exercitiul-1 -`);
+console.log(`- Collection of objects for all examples with cars -`);
+
+/*
+Collection of objects for all examples with cars
+*/
+
+const cars = [
+  {
+    make: "Honda",
+    model: "CR-V",
+    type: "suv",
+    amount: 14,
+    price: 24045,
+    onSale: true,
+  },
+  {
+    make: "Honda",
+    model: "Accord",
+    type: "sedan",
+    amount: 2,
+    price: 22455,
+    onSale: true,
+  },
+  {
+    make: "Mazda",
+    model: "Mazda 6",
+    type: "sedan",
+    amount: 8,
+    price: 24195,
+    onSale: false,
+  },
+  {
+    make: "Mazda",
+    model: "CX-9",
+    type: "suv",
+    amount: 7,
+    price: 31520,
+    onSale: true,
+  },
+  {
+    make: "Toyota",
+    model: "4Runner",
+    type: "suv",
+    amount: 19,
+    price: 34210,
+    onSale: false,
+  },
+  {
+    make: "Toyota",
+    model: "Sequoia",
+    type: "suv",
+    amount: 16,
+    price: 45560,
+    onSale: false,
+  },
+  {
+    make: "Toyota",
+    model: "Tacoma",
+    type: "truck",
+    amount: 4,
+    price: 24320,
+    onSale: true,
+  },
+  {
+    make: "Ford",
+    model: "F-150",
+    type: "truck",
+    amount: 11,
+    price: 27110,
+    onSale: true,
+  },
+  {
+    make: "Ford",
+    model: "Fusion",
+    type: "sedan",
+    amount: 13,
+    price: 22120,
+    onSale: true,
+  },
+  {
+    make: "Ford",
+    model: "Explorer",
+    type: "suv",
+    amount: 6,
+    price: 31660,
+    onSale: false,
+  },
+];
+
+/*
+Example 1 - Map method
+Let the `getModels` function return an array of models (model field) of all
+cars.
+*/
+
+console.log(`-Example 1 - Map method-`);
+
+const getModels = (cars) => {
+  return cars.map((x) => x.model);
+};
+
+console.table(getModels(cars));
+
+/*
+Example 2 - Map method
+Let the `makeCarsWithDiscount` function return a new array of objects with a changed
+value of the `price` property depending on the discount passed.
+*/
+
+console.log(`-Example 2 - Map method-`);
+
+const makeCarsWithDiscount = (cars, discount) => {
+  return cars.map((x) => x.price - discount * x.price);
+};
```

Comment:
This is not correct, you should return a new array of **objects with a changed
value of the `price` property**

Reply:
it is ok now ?

Reply:
const makeCarsWithDiscount = (cars, discount) => {
  return [...cars].map((x) => x.price - discount * x.price);
};

Reply:
but the value has been alreagy changed...!
I checked it out !